examples/star_unstar_a_gist.rs

```diff
@@ -0,0 +1,72 @@
+use octocrab::Octocrab;
+
+struct ProgramArguments {
+    gist_id: String,
+    star: bool,
+}
+
+/// Rudimentary CLI interface. Tries to be nice to the caller without too
+/// much bloat on the example.
+fn parse_argv_or_exit() -> ProgramArguments {
+    const USAGE: &str = r#"Usage:
+    <program> (--star | --unstar) GIST_ID"#;
+    let star = if let Some(param) = std::env::args().nth(1) {
+        if param == "--star" {
+            true
+        } else if param == "--unstar" {
+            false
+        } else {
+            eprintln!("error: Need (--star | --unstar) as first argument.");
+            eprintln!("{}", USAGE);
+            std::process::exit(1);
+        }
+    } else {
+        eprintln!("error: Need (--star | --unstar) as first argument.");
+        eprintln!("{}", USAGE);
+        std::process::exit(1);
+    };
+
+    let gist_id = if let Some(gist_id) = std::env::args().nth(2) {
+        gist_id
+    } else {
+        eprintln!("error: Need GIST_ID as second argument.");
+        eprintln!("{}", USAGE);
+        std::process::exit(1);
+    };
+    ProgramArguments { gist_id, star }
+}
```

Comment:
I've added this helper since I wasn't sure if pulling in a CLI dependency for a couple of arguments was a good idea. I don't have any strong opinions on these 3 endpoints living in the same example, happy to split them if this is considered too complex for an illustrative example.

Reply:
I don't have a strong opinion on examples as long they're clear, I also don't have a problem with using dependencies in them either.

The point of an example is to provide an minimal easily understandable version, anything that helps with that is encouraged.

Reply:
Thanks, noted. I'll keep that in mind for future examples!